---
string username
string password
string svr_hostname
string svr_ip